projects/ngx-translate-router/src/lib/gilsdav-reuse-strategy.ts

```diff
@@ -36,7 +36,11 @@ export class GilsdavReuseStrategy implements RouteReuseStrategy {
     if (!route.data.localizeRouter && (!route.parent || !route.parent.parent)) { // Lang route
       return 'LANG';
     } else if (route.data.localizeRouter) {
-      return `${this.getKey(route.parent)}/${route.data.localizeRouter.path}`;
+      let key = `${this.getKey(route.parent)}/${route.data.localizeRouter.path}`;
+      if (route.data.discriminantPathKey) {
+        key = `${route.data.discriminantPathKey}-${key}`;
```

Comment:
IMHO makes more sense to write the discriminant after the key:
 key += `-${route.data.discriminantPathKey}`;

EG: LANG/ROUTE1-DISCRIMINANT1/ROUTE2-DISCRIMINANT2